lib/modules/manager/cargo/extract.spec.ts

```diff
@@ -513,5 +519,40 @@ replace-with = "mcorbin"
       expect(res?.deps).toHaveLength(1);
       expect(res?.deps[0].packageName).toBe('boolector');
     });
+
+    it('extracts locked versions', async () => {
+      await writeLocalFile('Cargo.lock', cargo9lock);
+
+      const res = await extractPackageFile(cargo9toml, 'Cargo.toml', config);
+      expect(res?.deps[0].lockedVersion).toBe('2.0.1');
+      expect(res?.deps).toHaveLength(1);
+    });
+
+    it('extracts locked versions for renamed packages', async () => {
+      await writeLocalFile('Cargo.lock', cargo7lock);
+
+      const res = await extractPackageFile(cargo7toml, 'Cargo.toml', config);
+      expect(res?.deps[0].lockedVersion).toBe('1.0.1');
+      expect(res?.deps[1].lockedVersion).toBe('2.0.1');
+      expect(res?.deps).toHaveLength(2);
+    });
+
+    it('handles missing locked versions', async () => {
+      await writeLocalFile('Cargo.lock', cargo8lock);
+
+      const res = await extractPackageFile(cargo8toml, 'Cargo.toml', config);
+      expect(res?.deps[0].lockedVersion).toBe('2.0.1');
+      expect(res?.deps[1].lockedVersion).toBeUndefined();
+      expect(res?.deps).toHaveLength(2);
```

Comment:
same

Reply:
How do you expect me to match on non-presence of `lockedVersion` using `toMatchObject`?

This is not an improvement imo:
```javascript
expect(res?.deps).toMatchObject([
  { lockedVersion: '2.0.1' },
  expect.not.objectContaining({ lockedVersion: expect.anything() }),
]);
```

Reply:
If you use `toEqual()` then it must exactly match the contents

Reply:
Sure, but then it's asserting on fields that aren't really relevant to the test. I still feel the original code was fine.

Reply:
OK. I'm ok with toMatchObject() if you think some fields are irrelevant